src/dune_rules/melange.ml

```diff
@@ -0,0 +1,32 @@
+let melange_syntax =
+  Dune_lang.Syntax.create ~name:"melange" ~desc:"support for Melange compiler"
+    [ ((0, 1), `Since (3, 5)) ]
+
+module Cm_kind = struct
+  type t =
+    | Cmi
+    | Cmj
+
+  let source = function
+    | Cmi -> Ocaml.Ml_kind.Intf
+    | Cmj -> Impl
+
+  let ext = function
+    | Cmi -> ".cmi"
+    | Cmj -> ".cmj"
+
+  let to_dyn =
+    let open Dyn in
+    function
+    | Cmi -> variant "cmi" []
+    | Cmj -> variant "cmj" []
+
+  module Dict = struct
```

Comment:
Could you rename this `Dict` to `Map`.  I know we don't use this consistently, but we ended up deciding on `Map` in the codebase but didn't yet change all the existing code.

Reply:
https://github.com/jchavarri/dune/pull/4/commits/d1c3a8953cdfc4a4600c84bd9148b1a4a0dc0a29

Reply:
I thought we wanted to apply this change in all new modules, so replaced appearances in `Lib_mode` in https://github.com/jchavarri/dune/pull/4/commits/23c5b84c22e944f59ae338567a23230572527cd6.